### ISSUE TITLE
Switching between join display and mirror display causes number of workspaces to change and window tiling to become discombobulated 

### ISSUE THREAD

**Issue by davidparks21** [pop-os/pop#1666]
**Distribution (run `cat /etc/os-release`):**
 ```
NAME="Pop!_OS"
VERSION="20.10"
ID=pop
ID_LIKE="ubuntu debian"
PRETTY_NAME="Pop!_OS 20.10"
VERSION_ID="20.10"
HOME_URL="https://pop.system76.com"
SUPPORT_URL="https://support.system76.com"
BUG_REPORT_URL="https://github.com/pop-os/pop/issues"
PRIVACY_POLICY_URL="https://system76.com/privacy"
VERSION_CODENAME=groovy
UBUNTU_CODENAME=groovy
LOGO=distributor-logo-pop-os
```

**Issue/Bug Description:**
Open the `Displays` page in `Settings` and switch between `Join Displays` and `Mirror`, then optionally cancel that change and go back to `Join Displays`. This change switches from utilizing two monitors to one monitor. When this happens the number of workspaces changes and the layout of displays on the single monitor is buggy (some windows don't show at all, others are very oddly arranged).

**Steps to reproduce (if you know):**
Prerequisite: Tiling enabled; Dual monitor with Join Displays set; Windows on both monitors.
2nd monitor setup: Scale set to 200%; Fractional Scaling enabled
Open the `Displays` page in `Settings` and switch between `Join Displays` and `Mirror`, wait 10 seconds, then cancel the change.
Number of workspaces will have changed and tiling of windows is not orderly.

**Expected behavior:**
When switching from two monitors to one the number of workspaces should not change, windows should remain in the same workspace they were in before the change. Windows that were previously on two monitors should be cleanly re-tiled on the single display. 

Ideally, when cancelling the a update from `Join Displays` to `Mirror` the windows would re-tile to both displays as they were originally, though I would call this a stretch goal.

**Other Notes:**
Although this issue seems unlikely to be encountered often, I encounter it daily. The reason is that when I put the computer to sleep with 2 monitors connected, when I return from sleep mode the external monitor will not wake up automatically. The only solution I've found (other than rebooting) is to switch from `Join Displays` to `Mirror` (which properly starts the external monitor), and then simply cancelling the change to `Mirror` (at which point `Join Displays` returns with both monitors working). However when this sequence is executed the tiling and workspaces get totally discombobulated and require 5-10 minutes of work to reorganize every day.

On another note, there are a lot of issues with tiling in PopOS. This is the #1 area of concern I have overall in PopOS, so I encourage the dev team to tackle the many small issues there. Otherwise PopOS is heads and tails better than virtually every other desktop OS out there.

**Comment by huunghia160799** [comment]
This is also happening to me. It happens whenever I change from ```Join Displays``` to *one screen* only. It is the only thing bothering me from using Pop OS because I use a lot of windows and organizing them every time is tedious.

**Comment by DavidSmolinski** [comment]
I think I've had that bug since there were workspaces.
